File: app/src/sharedTest/java/org/metabrainz/mobile/AssertionUtils.java
package org.metabrainz.mobile;

import org.metabrainz.mobile.data.sources.api.entities.WikiSummary;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.Artist;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.Label;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.Recording;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.Release;
import org.metabrainz.mobile.data.sources.api.entities.mbentity.ReleaseGroup;

import static org.junit.Assert.assertEquals;

public class
AssertionUtils {

    public static void checkWikiAssertions(WikiSummary testSummary, WikiSummary summary) {
        assertEquals(testSummary.getExtract(), summary.getExtract());
    }

    public static void checkReleaseAssertions(Release testRelease, Release release) {
        assertEquals(testRelease, release);
        assertEquals(testRelease.getTitle(), release.getTitle());
        assertEquals(testRelease.getBarcode(), release.getBarcode());
        assertEquals(testRelease.getStatus(), release.getStatus());
        assertEquals(testRelease.getCountry(), release.getCountry());
        assertEquals(testRelease.getDisambiguation(), release.getDisambiguation());
        assertEquals(testRelease.getDate(), release.getDate());
    }

    public static void checkArtistAssertions(Artist testArtist, Artist artist) {
        assertEquals(artist, testArtist);
        assertEquals(testArtist.getCountry(), artist.getCountry());
        assertEquals(testArtist.getDisambiguation(), artist.getDisambiguation());
        assertEquals(testArtist.getName(), artist.getName());
        assertEquals(testArtist.getSortName(), artist.getSortName());
        assertEquals(testArtist.getGender(), artist.getGender());
    }

    public static void checkReleaseGroupAssertions(ReleaseGroup testReleaseGroup, ReleaseGroup releaseGroup) {
        assertEquals(testReleaseGroup, releaseGroup);
        assertEquals(testReleaseGroup.getTitle(), releaseGroup.getTitle());
        assertEquals(testReleaseGroup.getCount(), releaseGroup.getCount());
        assertEquals(testReleaseGroup.getPrimaryType(), releaseGroup.getPrimaryType());
        assertEquals(testReleaseGroup.getFullType(), releaseGroup.getFullType());
        assertEquals(testReleaseGroup.getDisambiguation(), releaseGroup.getDisambiguation());
    }

    public static void checkLabelAssertions(Label testLabel, Label label) {
        assertEquals(testLabel, label);
        assertEquals(testLabel.getCode(), label.getCode());
        assertEquals(testLabel.getType(), label.getType());
        assertEquals(testLabel.getCountry(), label.getCountry());
    }

    public static void checkRecordingAssertions(Recording testRecording, Recording recording) {
        assertEquals(testRecording, recording);
        assertEquals(testRecording.getTitle(), testRecording.getTitle());
        assertEquals(testRecording.getDuration(), recording.getDuration());
        assertEquals(testRecording.getLength(), recording.getLength());
        assertEquals(testRecording.getTrackCount(), recording.getTrackCount());
        assertEquals(testRecording.getDisambiguation(), recording.getDisambiguation());
    }

}

File: app/src/sharedTest/java/org/metabrainz/mobile/AssertionUtils.kt
package org.metabrainz.mobile

import org.hamcrest.MatcherAssert.assertThat
import org.junit.Assert.assertEquals
import org.hamcrest.collection.IsIterableContainingInAnyOrder.containsInAnyOrder
import org.metabrainz.mobile.data.sources.api.entities.WikiSummary
import org.metabrainz.mobile.data.sources.api.entities.mbentity.*
import org.metabrainz.mobile.data.sources.api.entities.mbentity.Collection
import org.metabrainz.mobile.presentation.features.adapters.ResultItem

object AssertionUtils {
    fun checkWikiAssertions(testSummary: WikiSummary, summary: WikiSummary) {
        assertEquals(testSummary.extract, summary.extract)
    }

    fun checkReleaseAssertions(testRelease: Release, release: Release) {
        assertEquals(testRelease, release)
        assertEquals(testRelease.title, release.title)
        assertEquals(testRelease.barcode, release.barcode)
        assertEquals(testRelease.status, release.status)
        assertEquals(testRelease.country, release.country)
        assertEquals(testRelease.disambiguation, release.disambiguation)
        assertEquals(testRelease.date, release.date)
        if (testRelease.artistCredits.size != 0) assertThat(
            testRelease.artistCredits,
            containsInAnyOrder(release.artistCredits.toTypedArray())
        )
    }

    fun checkArtistAssertions(testArtist: Artist, artist: Artist) {
        assertEquals(testArtist, artist)
        assertEquals(testArtist.country, artist.country)
        assertEquals(testArtist.disambiguation, artist.disambiguation)
        assertEquals(testArtist.name, artist.name)
        assertEquals(testArtist.sortName, artist.sortName)
        assertEquals(testArtist.gender, artist.gender)
        assertEquals(testArtist.type, artist.type)
        if (testArtist.getReleases().isNotEmpty()) assertThat(
            testArtist.getReleases(),
            containsInAnyOrder(artist.getReleases().toTypedArray())
        )
    }

    fun checkReleaseGroupAssertions(testReleaseGroup: ReleaseGroup, releaseGroup: ReleaseGroup) {
        assertEquals(testReleaseGroup, releaseGroup)
        assertEquals(testReleaseGroup.title, releaseGroup.title)
        assertEquals(testReleaseGroup.count, releaseGroup.count)
        assertEquals(testReleaseGroup.primaryType, releaseGroup.primaryType)
        assertEquals(testReleaseGroup.fullType, releaseGroup.fullType)
        assertEquals(testReleaseGroup.disambiguation, releaseGroup.disambiguation)
        if (testReleaseGroup.artistCredits.size != 0) assertThat(
            testReleaseGroup.artistCredits,
            containsInAnyOrder(releaseGroup.artistCredits.toTypedArray())
        )
    }

    fun checkLabelAssertions(testLabel: Label, label: Label) {
        assertEquals(testLabel, label)
        assertEquals(testLabel.code, label.code)
        assertEquals(testLabel.type, label.type)
        assertEquals(testLabel.country, label.country)
    }

    fun checkRecordingAssertions(testRecording: Recording, recording: Recording) {
        assertEquals(testRecording, recording)
        assertEquals(testRecording.title, testRecording.title)
        assertEquals(testRecording.duration, recording.duration)
        assertEquals(testRecording.length, recording.length)
        assertEquals(testRecording.trackCount, recording.trackCount)
        assertEquals(testRecording.disambiguation, recording.disambiguation)
        if (testRecording.artistCredits.size != 0) assertThat(
            testRecording.artistCredits,
            containsInAnyOrder(recording.artistCredits.toTypedArray())
        )
    }

    fun checkResultItemAssertions(testItem: ResultItem, item: ResultItem) {
        assertEquals(testItem.mBID, item.mBID)
        assertEquals(testItem.disambiguation, item.disambiguation)
        assertEquals(testItem.name, item.name)
        assertEquals(testItem.primary, item.primary)
        assertEquals(testItem.secondary, item.secondary)
    }

    fun checkCollectionDetailsAssertions(testCollectionDetails: ResultItem, collectionDetails: ResultItem) {
        assertEquals(testCollectionDetails, collectionDetails)
        assertEquals(testCollectionDetails.name, collectionDetails.name)
        assertEquals(testCollectionDetails.disambiguation, collectionDetails.disambiguation)
        assertEquals(testCollectionDetails.name, collectionDetails.name)
        assertEquals(testCollectionDetails.name, collectionDetails.name)
    }

    fun checkCollectionAssertions(testCollection: Collection, collection: Collection) {
        assertEquals(testCollection, collection)
        assertEquals(testCollection.mbid, collection.mbid)
        assertEquals(testCollection.type, collection.type)
        assertEquals(testCollection.name, collection.name)
        assertEquals(testCollection.count, collection.count)
        assertEquals(testCollection.editor, collection.editor)
        assertEquals(testCollection.entityType, collection.entityType)
    }

}